libs/logic-apps-shared/src/designer-client-services/lib/experimentation.ts

```diff
@@ -0,0 +1,24 @@
+import { AssertionException, AssertionErrorCode } from '../../utils/src';
+
+export interface IExperimentationService {
+  isFeatureEnabled: (featureGateName: string, type: 'extension' | 'shell') => Promise<boolean>;
+  // Setting the return as any since depending on the feature, you can return a string, number or a boolean
+  getFeatureValue: (featureGateName: string, type: 'extension' | 'shell') => Promise<any>;
+}
+
+let service: IExperimentationService;
+
+export const InitExperimentationServiceService = (experimentationService: IExperimentationService): void => {
+  service = experimentationService;
+};
+
+export const ExperimentationService = (): IExperimentationService => {
+  if (!service) {
```

Comment:
Just to be extra sure about this. The designer doesn't crash or throws this error if the service is not needed and not initilialized, right? for example in vscode extension

Reply:
It shouldn't. The designer will only crash in scenarios where the experimentation service is being used but not initialized. Each client using the designer needs to initialize the service. On the side I'll also be checking to see how we can always default to a certain value value here

Reply:
Actually, I went ahead and added it explicitly for vscode as well. We need a better way to default certain service if the value remains the same for all default experiences.

Reply:
On a second thought, I went ahead and always defaulted to a false implementation of ExperimentationService so incase a certain client using the designer does not want to use the A/B test feature, they won't get it unless they explicitly add in the experimentation service

Reply:
Thanks Krrish!